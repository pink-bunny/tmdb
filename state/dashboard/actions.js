import * as types from './types';

export const fetchTrendingMovies = (page = 1) => ({
  type: types.FETCH_TRENDING_MOVIES,
  page
});

export const fetchTrendingMoviesSuccess = (ids, totalItems, currentPage) => ({
  type: types.FETCH_TRENDING_MOVIES_SUCCESS,
  ids,
  totalItems,
  currentPage
});

export const fetchTrendingMoviesError = (error) => ({
  type: types.FETCH_TRENDING_MOVIES_ERROR,
  error
});

export const searchMovies = ({ search }, setErrors, setSubmitting, page = 1) => ({
  type: types.SEARCH_MOVIES,
  search,
  page,
  setErrors,
  setSubmitting
});
