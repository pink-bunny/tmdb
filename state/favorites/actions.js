import * as types from './types';

export const fetchFavorites = (page = 1) => ({
  type: types.FETCH_FAVORITES,
  page
});
export const fetchFavoritesSuccess = (ids, totalItems, currentPage) => ({
  type: types.FETCH_FAVORITES_SUCCESS,
  ids,
  totalItems,
  currentPage
});

export const fetchFavoritesError = (error) => ({
  type: types.FETCH_FAVORITES_ERROR,
  error
});

export const toggleToFavorite = (id, favorite, needRefetchList = false) => ({
  type: types.TOGGLE_TO_FAVORITES,
  id,
  favorite,
  needRefetchList
});

export const toggleToFavoriteSuccess = (id, isMovieInFavorites) => ({
  type: types.TOGGLE_TO_FAVORITES_SUCCESS,
  id,
  isMovieInFavorites
});
