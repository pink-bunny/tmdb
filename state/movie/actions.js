import * as types from './types';

export const fetchMovie = (id) => ({
  type: types.FETCH_MOVIE,
  id
});

export const fetchMovieSuccess = (id) => ({
  type: types.FETCH_MOVIE_SUCCESS,
  id
});

export const fetchMovieError = (error) => ({
  type: types.FETCH_MOVIE_ERROR,
  error
});

export const clearMovieFromState = () => ({
  type: types.CLEAR_MOVIE_FROM_STATE
});
