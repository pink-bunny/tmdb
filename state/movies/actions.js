import * as types from './types';

export const fetchTrendingMovies = (pageNum = 1) => ({
  type: types.FETCH_TRANDING_MOVIES,
  pageNum
});

export const fetchTrendingMoviesSuccess = (list, totalPages) => ({
  type: types.FETCH_TRANDING_MOVIES_SUCCESS,
  list,
  totalPages
});

export const fetchTrendingMoviesError = (error) => ({
  type: types.FETCH_TRANDING_MOVIES_ERROR,
  error
});
