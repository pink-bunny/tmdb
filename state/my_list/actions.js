import * as types from './types';

export const fetchMyList = (id) => ({
  type: types.FETCH_MY_LIST,
  id
});

export const fetchMyListSuccess = (movieId) => ({
  type: types.FETCH_MY_LIST_SUCCESS,
  movieId
});

export const fetchMyListError = (error) => ({
  type: types.FETCH_MY_LIST_ERROR,
  error
});
