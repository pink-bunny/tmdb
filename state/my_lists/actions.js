
import * as types from './types';

export const fetchMyLists = () => ({
  type: types.FETCH_MY_LISTS
});

export const fetchMyListsSuccess = (ids) => ({
  type: types.FETCH_MY_LISTS_SUCCESS,
  ids
});

export const fetchMyListsError = (error) => ({
  type: types.FETCH_MY_LISTS_ERROR,
  error
});
