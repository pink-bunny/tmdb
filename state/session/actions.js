import * as types from './types';

export const sessionRequestToken = () => ({
  type: types.SESSION_REQUEST_TOKEN
});

export const sessionRequestTokenSuccess = (token) => ({
  type: types.SESSION_REQUEST_TOKEN_SUCCESS,
  token
});
