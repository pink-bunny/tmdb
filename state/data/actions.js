import * as types from './types';
import { TOGGLE_TO_WATCHLIST_SUCCESS } from '../movie/types';

export const mergeMoviesList = (movies) => ({
  type: types.MERGE_MOVIES_LIST,
  movies
});

export const mergeListsList = (lists) => ({
  type: types.MERGE_LISTS_LIST,
  lists
});

export const toggleToWatchlistSuccess = (id, isMovieInWatchlist) => ({
  type: TOGGLE_TO_WATCHLIST_SUCCESS,
  id,
  isMovieInWatchlist
});
