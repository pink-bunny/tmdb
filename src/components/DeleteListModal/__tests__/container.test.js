import React from 'react';
import configureStore from 'redux-mock-store';
import { shallow } from 'enzyme';

import ConnectedDeleteListModal from '../container';
import { deleteMyList } from '../../../../../state/my_list/actions';

jest.mock('../../../../../state/my_list/actions', () => ({
  deleteMyList: jest.fn()
}));

jest.mock('../../../../../state/my_lists/selectors', () => ({
  myListsDeleteListError: jest.fn(() => '')
}));

describe('DeleteListModalContainer', () => {
  const store = configureStore()({});
  const props = {
    store,
    name: 'Mock list',
    id: 1
  };
  store.dispatch = jest.fn();
  /* eslint-disable react/jsx-props-no-spreading */
  const wrapper = shallow(<ConnectedDeleteListModal {...props} />).dive().dive().dive();
  /* eslint-enable react/jsx-props-no-spreading */
  const instance = wrapper.instance();

  it('matches snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('deleteList method called with params', () => {
    wrapper.setProps({ hideModal: jest.fn() });
    const { id, hideModal } = wrapper.props();

    instance.deleteList();
    expect(deleteMyList).toHaveBeenCalledWith(id, hideModal);
  });
});
