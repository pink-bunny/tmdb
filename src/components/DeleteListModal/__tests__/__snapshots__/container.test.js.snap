// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DeleteListModalContainer matches snapshot 1`] = `
<DeleteListModal
  deleteList={[Function]}
  error=""
  handleTriggerClick={[Function]}
  hideModal={[MockFunction]}
  id={1}
  modalVisible={false}
  name="Mock list"
  triggerComponent={[MockFunction]}
  triggerProps={null}
/>
`;
